org.adempiere.base/src/org/compiere/util/Ini.java

```diff
@@ -794,4 +798,76 @@ public static String getPropertyFileName()
 	{
 		return s_propertyFileName;
 	}
+
+	public static String getVar(String secretVar) {
+		String cmd = getUtilsCmd("getVar");
+		String[] command = new String[] {
+				cmd,
+				secretVar
+		};
+		String retValue = runCommand(command);
+		return retValue;
+	}
+
+	public static void setVar(String secretVar, String secretValue) {
+		String cmd = getUtilsCmd("setVar");
+		String[] command = new String[] {
+				cmd,
+				secretVar,
+				secretValue
+		};
+		@SuppressWarnings("unused") // used for debugging purposes
+		String retValue = runCommand(command);
+	}
+
+	private static String getUtilsCmd(String script) {
+		File utilsFolder = new File(getAdempiereHome() + File.separator + "utils");
+		if (! utilsFolder.exists()) {
+			// /utils does not exist, probably running on eclipse
+			if (Env.isWindows()) {
+				utilsFolder = new File(getAdempiereHome() + File.separator + "org.adempiere.server-feature" + File.separator + "utils.windows");
+			} else {
+				utilsFolder = new File(getAdempiereHome() + File.separator + "org.adempiere.server-feature" + File.separator + "utils.unix");
+			}
+			if (! utilsFolder.exists()) {
+				throw new AdempiereException("Folder utils does not exist");
+			}
+		}
+		File cmd = new File(utilsFolder, script + (Env.isWindows() ? ".bat" : ".sh"));
+		if (! cmd.exists() || ! cmd.canExecute()) {
+			throw new AdempiereException("File does not exist or canno execute " + cmd.getAbsolutePath());
+		}
+		return cmd.getAbsolutePath();
+	}
+
+	public static String runCommand(String[] command) {
+		StringBuilder msg = new StringBuilder();
+		try {
+			String s;
+			Process p = Runtime.getRuntime().exec(command);
+			BufferedReader stdInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+			BufferedReader stdError = new BufferedReader(new InputStreamReader(p.getErrorStream()));
+			// read the output from the command
+			while ((s = stdInput.readLine()) != null) {
+				msg.append(s);
+			}
+			// read any errors from the attempted command
+			while ((s = stdError.readLine()) != null) {
+				msg.append(s);
+			}
+			try {
+				p.waitFor(5, TimeUnit.SECONDS);
+			} catch (InterruptedException e) {

```

Comment:
from API of Process.waitFor
> Returns:
true if the process has exited and false if the waiting time elapsed before the process has exited.
Throws:
InterruptedException - if the current thread is interrupted while waiting.

so message "Timeout waiting for" isn't correct for InterruptedException

by above mistake then next check "if (p.exitValue() != 0) {" too soon process can be not yet done when code run to there